packages/react-vapor/src/components/editor/JSONEditor.tsx

```diff
@@ -5,73 +5,76 @@ import * as React from 'react';
 
 import {Svg} from '../svg/Svg';
 import {CodeEditor} from './CodeEditor';
-import {CodeMirrorModes} from './EditorConstants';
+import {CodeMirrorModes, DEFAULT_JSON_ERROR_MESSAGE} from './EditorConstants';
+import {JSONEditorUtils} from './JSONEditorUtils';
 
 export interface IJSONEditorProps {
+    id?: string;
     value: string;
     readOnly?: boolean;
     onChange?: (json: string, inError: boolean) => void;
     errorMessage?: string;
+    containerClasses?: string[];
+    className?: string;
+    ref?: React.Ref<any>;
 }
 
-export interface IJSONEditorState {
-    isInError: boolean;
+export interface IJSONEditorStateProps {

```

Comment:
Are we not refraining from prefixing interfaces with `I`? I know it breaks the pattern in this case, so not sure what's best practice 🤔

Reply:
I personally have no preference, it seems that using `I` for the naming of interfaces is the pattern existed in react-vapor :thinking:

Reply:
[Link](https://til.hashrocket.com/posts/cfnlyzxcrc-do-not-prefix-typescript-interface-names) I think we should just drop the I